SPLICE-1640 apply memory limit check for consecutive outer broadcast …(2.5)

yxia92 opened the pull request:
…join and derived tables

dgomezferro commented on db-engine/src/main/java/com/splicemachine/db/iapi/sql/compile/Optimizable.java:
```diff
@@ -428,4 +428,9 @@ boolean feasibleJoinStrategy(OptimizablePredicateList predList,
+	/**
+	 * get the current optimizable's memory usage if its best join plan is a broadcast join
+	 */
+	double getMemoryUsage4BCJ();
```
I'd rather have a complete name, rather than the abbreviature. I wouldn't mind on a local variable but for the method name I think it's better to spell it out.

yxia92 commented:
@dgomezferro , I refactored the function name getMemoryUsage4BCJ() to getMemoryUsage4BroadcastJoin() based on your comment and updated the full request.
